golem-cli/src/oss/clients/component.rs

```diff
@@ -116,17 +142,35 @@ impl<C: golem_client::api::ComponentClient + Sync + Send> ComponentClient
         urn: ComponentUrn,
         path: PathBufOrStdin,
         component_type: Option<golem_client::model::ComponentType>,
+        files_archive: Option<&Path>,
+        files_permissions: Option<&ComponentFilePathAndPermissionsList>,
     ) -> Result<Component, GolemError> {
         info!("Updating component {urn} from {path:?}");
 
+        let files_archive_file = match files_archive {
+            Some(fa) => {
+                let file = File::open(fa)
+                    .await
+                    .map_err(|e| GolemError(format!("Can't open component files archive: {e}")))?;
+                Some(file)
+            }
+            None => None,
+        };
```

Comment:
If you like it, I think `map` and `transpose` would make it feel less `duplicated` between `add` and `update`

Reply:
same as above